samples/cpp/benchmark_app/main.cpp

```diff
@@ -938,7 +944,18 @@ int main(int argc, char* argv[]) {
                 // method of `std::exception` So, rechecking for any exceptions here.
                 inferRequest->wait();
                 inferRequest->start_async();
-            }
+
+                // setting target duration allow to get required number of FPS.
+                // if number of required FPS is not set it will be calculated from reported duration and number of
+                // iterations.
+                if (fps_number > 0) {
+                    double targetDuration = double(batchSize * 1000 / fps_number);
```

Comment:
This code would not work for dynamic case (when batchSize is dynamically changed)